src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/VisitRequestsService.kt

```diff
@@ -91,13 +93,26 @@ class VisitRequestsService(
     LOG.info("Entered VisitRequestsService - autoRejectRequestVisitsAtMinimumBookingWindow")
 
     val requestVisitsDueForAutoRejection = visitRepository.findAllVisitRequestsDueForAutoRejection()
+    return processAutoRejectRequestVisits(requestVisitsDueForAutoRejection, rejectionText = "Auto rejected by minimum booking window system cron")
+  }
+
+  fun handlePrisonerReleasedEventAutoRejectRequestVisits(notificationDto: PrisonerReleasedNotificationDto): Int {
+    LOG.info("Entered VisitRequestsService - handlePrisonerReleasedEventAutoRejectRequestVisits")
 
+    val requestVisitsDueForAutoRejection = visitRepository.findAllVisitRequestsForPrisoner(notificationDto.prisonerNumber)
+    return processAutoRejectRequestVisits(requestVisitsDueForAutoRejection, rejectionText = "Auto rejected by prisoner released event")
+  }
+
+  private fun processAutoRejectRequestVisits(requestVisitsDueForAutoRejection: List<Visit>, rejectionText: String): Int {
```

Comment:
suggestion only - do we want rejectionText to come from an enum? This type could also be sent to AppInsights if needed for tracking why visits have been auto rejected.